### ISSUE TITLE
Errors when validating model output accuracy during export

### ISSUE THREAD

**Issue by SebastianJanampa** [onnx/onnx#8371]
### Describe the bug
I am trying to export D-FINE-m to ONNX using Google Colab and the original [D-FINE repo.](https://github.com/Peterande/D-FINE/blob/master) The export works, but the report says there's an issue when it validates model output accuracy. 

### System information
Google Collab

### Reproduction instructions
- I only did a modification of the exportation code found in [`tools/deployment/export_onnx.py`](https://github.com/Peterande/D-FINE/blob/master/tools/deployment/export_onnx.py) starting in line 71.

```python
    torch.onnx.export(
        model,
        (data, size),
        output_file,
        input_names=["images", "orig_target_sizes"],
        output_names=["labels", "boxes", "scores"],
        dynamic_axes=dynamic_axes,
        opset_version=18,
        verbose=False,
        do_constant_folding=True,
        dynamo=True,
        external_data=False,
        verify=True,
        report=True,
    )
```

### Notes
I used the ONNX model, and it works well even though the report shows large absolute and relative differences. How could I solve that? Also, could you share resources to go from zero to hero on ONNX?

Best,
Sebastian

**Comment by SebastianJanampa** [comment]
Here is the [report](https://github.com/user-attachments/files/31532985/onnx_export_2026-08-27_21-22-26-630078_accuracy.md)

**Comment by Ankita7033** [comment]
I'd like to work on this issue.
My hypothesis: the large diffs likely come from floating-point divergence in the deformable attention/sampling ops, or near-tied class logits causing argmax/topk to pick different indices between PyTorch and ONNX Runtime — both common with dynamo=True and not necessarily real bugs.
Plan: reproduce with dynamo=False to isolate whether it's exporter-specific, then compare decoded final boxes/scores/labels (not raw tensor allclose) to check actual detection correctness. Will follow up with findings before opening a PR.
